### PR TITLE
Home page banner

### PR DESCRIPTION
**Pull request by nibivi77**
# Description

[Landing page hero image and title] https://trello.com/c/QWkOCj4B/57-landing-page-hero-image-and-title

Added a banner to look as close as possible to the https://www.canada.ca/en.html banner.

For the hard-coded strings as English, we will need to change it when i18n is incorporated into the project.

## Acceptance Criteria
Content AC:
 - consists of a title, followed by short byline
 - the site title and byline appear in a dark blue box
 - the site title is presented as an H1, with a red horizontal rule underneath
 - the byline appears immediately below the site title

Image AC:
 - images must be photographic (no illustration or abstract imagery)
 - choose images that show seasonal details (e.g. snow in winter, flowers in spring)
 - avoid images of people or faces as these are distracting
 - the banner image appears across the full width of the page, underneath the site header and behind the box containing the site title and headline

## Test Instructions

1. Navigate to branch called: homePageBanner
2. runthe comand: npm run dev
3.  see the banner on the home page.
4. Test different width including mobile.

## Help Requested

Any feedback is greatly appreciated. I am new to JavaScript unit test and Storybook.

## Checklist

- [ ] Strings use placeholders for translation (No hard coded strings)
- [ x] Unit tests have been added/updated

## Product and Sprint Backlog

https://trello.com/c/QWkOCj4B/57-landing-page-hero-image-and-title

### CODE REVIEW

File: components/atoms/Banner.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+/**
+ * Displays the banner on the page
+ */
+
+export const Banner = ({ siteTitle, headLine }) => {
```
Comment (pcraig3) [inline components/atoms/Banner.js:8]: Can we say `headline` (lowercase `L`)>

**Review by pcraig3 — COMMENTED** [review]
Two small feedbacks.

1. When the overlay hits the left-hand side of the screen, it should be full-bleed 


Screen Shot 2021-03-24 at 10 33 55 AM

It's the orange part there that's missing. I get that as the screen squeezes down it will touch both sides of the screen, but we should just force it to do that.

2. Can you do a screenshot of your change in your PR description since it's a visual change, in future?

File: components/atoms/Banner.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+Banner.propTypes = {
+  /**
+   * Text area that displays the title for the page
+   */
+  siteTitle: PropTypes.string.isRequired,
+
+  /**
+   * This is for adding an id for testing in cypress
+   */
+  headLine: PropTypes.string.isRequired,
```
Comment (pcraig3) [inline components/atoms/Banner.js:29]: have to update this too

File: components/atoms/Banner.stories.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  component: Banner,
+};
+
+const Template = (args) => <Banner {...args} />;
+
+export const Primary = Template.bind({});
+
+Primary.args = {
+  siteTitle: "Banner Text",
+  headLine: "Banner Headline",
```
Comment (pcraig3) [inline components/atoms/Banner.stories.js:15]: and update this

File: tailwind.config.js [hunk 4]
```diff
@@ -2,7 +2,15 @@ module.exports = {
...
   darkMode: false, // or 'media' or 'class'
   theme: {
-    extend: {},
+    extend: {
+      backgroundImage: theme => ({
+        'banner-img' :"url('../../public/bg-banner.jpg')"
+      }),
+      colors: {
+        'dk-blue': '#4F5D6C',
+        'hr-red-bar': '#BD6166'
```
Comment (pcraig3) [inline tailwind.config.js:11]: Sorry to go back and forth on this: can we match the Canada.ca colours for the background colour on the box?

**Review by pcraig3 — COMMENTED** [review]
Screen Shot 2021-03-24 at 11 10 42 AM

I feel it's too long on the medium max size, but we can leave it for now (until we have a better site layout). eg, on https://www.canada.ca/en.html, it never gets that big.

**Comment by nibivi77** [comment]
> Screen Shot 2021-03-24 at 11 10 42 AM
> 
> I feel it's too long on the medium max size, but we can leave it for now (until we have a better site layout). eg, on https://www.canada.ca/en.html, it never gets that big.

This will be adjusted once we have left and right margin for with layout.

File: components/atoms/Banner.test.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+
+import { Primary } from "./Banner.stories";
+
+it("renders Banner in its primary state", () => {
```
Comment (will0684) [inline components/atoms/Banner.test.js:7]: We want to do a11y checks with jest-axe as well. Take a look at the home.spec.js to see what that should look like, it is very simple. You will want to add a "describe" block as well to wrap the tests like in the home.spec.js

**Review by will0684 — CHANGES_REQUESTED** [review]
Responsive and no strange behaviours nice work. I added a request related to the testing.

File: components/atoms/Banner.test.js (hunk 5 above)
Comment (will0684) [reply components/atoms/Banner.test.js:7]: Additionally you will want to put a test with jest-axe in components/atoms/Banner.test.js like this:

https://github.com/nickcolley/jest-axe#testing-react-with-react-testing-library

**Review by will0684 — APPROVED** [review]

**Comment by nibivi77** [comment]
> Two small feedbacks.
> 
> 1. When the overlay hits the left-hand side of the screen, it should be full-bleed
> 
> Screen Shot 2021-03-24 at 10 33 55 AM
> 
> It's the orange part there that's missing. I get that as the screen squeezes down it will touch both sides of the screen, but we should just force it to do that.
> 
> 1. Can you do a screenshot of your change in your PR description since it's a visual change, in future?

Banner request fixed and I will include a screenshot for future visual PR's :)

File: components/atoms/Banner.js (hunk 1 above)
Comment (nibivi77) [reply components/atoms/Banner.js]: changes made.

File: components/atoms/Banner.js (hunk 2 above)
Comment (nibivi77) [reply components/atoms/Banner.js]: Changes made.

File: components/atoms/Banner.stories.js (hunk 3 above)
Comment (nibivi77) [reply components/atoms/Banner.stories.js]: Changes made.

File: tailwind.config.js (hunk 4 above)
Comment (nibivi77) [reply tailwind.config.js]: Changes made.

**Pull request merged**